pos_localstorage/static/src/js/models.js

```diff
@@ -0,0 +1,224 @@
+                    // If this fails, we load the information from the server.
+                    var product = pos.db.load('product_' + products[i].id);
+                    if (product) {
+                        products_to_add.push(product);
+                    } else {
+                        products_to_fetch.push(products[i].id);
+                    }
+                }
+                // Add products that we have in localStorage
+                pos.db.add_products(products_to_add);
+
+                // Fetch all products that we don't have the data
+                //  because they were not in localStorage or they are new
+                return pos.fetch('product.product',
+                                 productAttributeList,
+                                 ['|', ['id', 'in', products_to_fetch],
+                                       '&', '&', ['sale_ok', '=', true], ['available_in_pos', '=', true], ['write_date', '>=', lastUpdate]],
+                                 {pricelist: pos.get('shop').pricelist_id[0]} // context for price
+                );
+            }).then(function(products){
+                pos.db.add_products(products);
+
+                lastUpdate = new Date(Date.now() - _searchGracePeriod).toISOString();
+                pos.db.save('last_product_update', lastUpdate);
+            });
+        },
+    };
+
+
+    var _old_initialize = module.PosModel.prototype.initialize;
+    module.PosModel.prototype.initialize = function(session, attributes) {
+        var ret = _old_initialize.call(this, session, attributes);
+        // HACK to make localStorage abstraction dependent of the database
+        this.db.name = session.db;
+        // clear database again after setting the name;
+        this.db.clear('products','categories');
+        return ret;
+    };
+
+    // Overwrite method to additionally save product data in localStorage
+    var _old_add_products = module.PosLS.prototype.add_products;
+    module.PosLS.prototype.add_products = function(products) {
+        _old_add_products.call(this, products);
+        try {
+            for (var i in products) {
+                var product = products[i];
+                this.save('product_' + product.id, product);
+            }
+        } catch(e) {
+            if (console) {
+                console.log("Could not store all products into LocalStorage");
+                console.log(e);
+            }
+            // We delete the last ~100 products to free up space.
+            // If localStorage is full, some other features may crash.
+            var length = localStorage.length;
+            for (var i = 1; i <= 100; ++i) {
+                var key = localStorage.key(length - i);
+                if (/product_\d+$/.exec(key)) {
+                    localStorage.removeItem(key);
+                }
+            }
+        }
+    };
+
+    // overwrite method in order to replace product loading
```

Comment:
Not a big deal.
V7 OCA module are not very used I guess. (because V7 Odoo-POS neither...) My PoV : 
A single warning in your openerp file is sufficient.
If people are interested to solve this problem, it is possible to write a glue module that depend of both modules, and to fix the problem.